carapace-server/src/main/java/org/carapaceproxy/server/mapper/StandardEndpointMapper.java

```diff
@@ -219,27 +215,9 @@ public void configure(ConfigurationStore properties) throws ConfigurationNotVali
             if (!id.isEmpty()) {
                 String action = properties.getProperty(prefix + "action", "");
                 boolean enabled = Boolean.parseBoolean(properties.getProperty(prefix + "enabled", "false"));
-                String match = properties.getProperty(prefix + "match", "all").trim();
-                int space = match.indexOf(' ');
-                String matchType = match;
-                if (space >= 0) {
-                    matchType = match.substring(0, space);
-                    match = match.substring(space + 1);
-                }
-                RequestMatcher matcher;
-                switch (matchType) {
-                    case "all":
-                        matcher = new MatchAllRequestMatcher();
-                        break;
-                    case "regexp":
-                        matcher = new URIRequestMatcher(match);
-                        break;
-                    default:
-                        throw new ConfigurationNotValidException(prefix + "match can be only 'all' and 'regexp' at the moment");
-
-                }
-                LOG.log(Level.INFO, "configured route {0} action: {1} enabled: {2} matcher: {3}", new Object[]{id, action, enabled, matcher});
-                RouteConfiguration config = new RouteConfiguration(id, action, enabled, matcher);
+                String matchingCondition = properties.getProperty(prefix + "match", "all").trim();
```

Comment:
Here I am expecting to create the RequestMatcher bu parsing the configuration value, this is to be done only once.
And a parse error will be treated as a invalid configuration error, not a runtime error